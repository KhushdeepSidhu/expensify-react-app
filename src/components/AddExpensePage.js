import React from 'react'

const AddExpensePage = () => (
    <div>
        <p>This is from the create expense component.</p>
    </div>
)

export { AddExpensePage as default } 