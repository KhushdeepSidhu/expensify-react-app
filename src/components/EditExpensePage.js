import React from 'react'
import { connect } from 'react-redux'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

export class EditExpensePage extends React.Component {

    onSubmit = ( expense ) => {
        this.props.startEditExpense ( { id: this.props.expense.id, updates: expense } )
        this.props.history.push ( '/' )
    }

    onClick = () => {
        this.props.startRemoveExpense ( { id: this.props.expense.id } )
        this.props.history.push ( '/' )
    }

    render () {
        return (
            <div>
                <div className = "page-header">
                    <div className = "content-container">
                        <h2 className = "page-header__title">Edit Expense</h2>
                    </div>
                </div>
                <div className = "content-container">
                    <ExpenseForm 
                        onSubmit = { this.onSubmit }
                        expense = { this.props.expense }
                    />
                    <button 
                        className = "button button--remove"
                        onClick = { this.onClick } 
                    >
                    Remove Expense
                    </button>
                </div>
            </div>
        )
    }

}


const mapStateToProps = ( state, props ) => ( {
    expense: state.expenses.find ( ( expense ) => expense.id === props.match.params.id )
} )

const mapDispatchToProps = ( dispatch ) => ( {
    startEditExpense: ( { id, updates } ) => dispatch ( startEditExpense ( { id, updates } ) ),
    startRemoveExpense: ( { id } ) => dispatch ( startRemoveExpense ( { id } ) )
} )


export default connect ( mapStateToProps, mapDispatchToProps ) ( EditExpensePage )
