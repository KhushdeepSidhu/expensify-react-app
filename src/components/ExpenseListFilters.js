import React from 'react'
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    }

    onDatesChange = ( { startDate, endDate } ) => {
        this.props.setStartDate ( startDate )
        this.props.setEndDate ( endDate )
    }

    onFocusChange = (  calendarFocused  ) => {
        this.setState ( () => ( { calendarFocused } ) )
    }

    onTextChange = ( event ) => {
        this.props.setTextFilter ( event.target.value )
    }

    onSortByChange = ( event ) => {
        if ( event.target.value === 'amount' ) {
            this.props.sortByAmount()
        } else if ( event.target.value === 'date' ) {
            this.props.sortByDate()
        }
    }

    render () {

        return (
            <div className = "content-container">
                <div className = "input-group">
                    <div className = "input-group__item">
                        <input 
                            className = "text-input"
                            placeholder = "Search expenses..."
                            type = "text" 
                            onChange = { this.onTextChange }
                            value = { this.props.filters.text } 
                        />
                    </div>
                    <div className = "input-group__item">
                        <select 
                            className = "select"
                            value = { this.props.filters.sortBy }
                            onChange = { this.onSortByChange }
                        >
                            <option value = "date" >Date</option>
                            <option value = "amount">Amount</option>
                        </select>
                    </div>
                    <div className = "input-group__item">
                        <DateRangePicker
                            startDate = { this.props.filters.startDate }
                            startDateId = "start-date-id"
                            endDate = { this.props.filters.endDate }
                            endDateId = "end-date-id"
                            onDatesChange = { this.onDatesChange }
                            focusedInput = { this.state.calendarFocused }
                            onFocusChange = { this.onFocusChange }
                            numberOfMonths = { 1 }
                            isOutsideRange = { () => false }
                            showClearDates = { true }
                        />
                    </div>
                </div>
            </div>
        )

    }

}

const mapDispatchToProps = ( dispatch ) => ( {
    setStartDate: ( startDate ) => dispatch ( setStartDate ( startDate ) ),
    setEndDate: ( endDate ) => dispatch ( setEndDate ( endDate ) ),
    setTextFilter: ( text ) => dispatch ( setTextFilter ( text ) ),
    sortByAmount: () => dispatch ( sortByAmount () ),
    sortByDate: () => dispatch ( sortByDate () )
} )

const mapStateToProps = ( state ) => {
    return {
        filters: state.filters
    }
}

export default connect ( mapStateToProps, mapDispatchToProps ) ( ExpenseListFilters )