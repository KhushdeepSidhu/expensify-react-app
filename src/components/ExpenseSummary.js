import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'

export const ExpenseSummary = ( { expensesCount, expensesTotal } ) => {

    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = numeral ( expensesTotal / 100 ).format ( '$0,0.00' )

    return (
        <div>
            <h3>
                Viewing { expensesCount } { expenseWord } totalling { formattedExpensesTotal }
            </h3>
        </div>
    )

}

const mapStateToProps = ( state ) => {
    const visibleExpenses = selectExpenses ( state.expenses, state.filters )
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal ( visibleExpenses )  
    }
}

export default connect ( mapStateToProps ) ( ExpenseSummary )