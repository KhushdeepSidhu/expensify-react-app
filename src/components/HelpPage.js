import React from 'react'

const HelpPage = () => (
    <div>
        <p>This is from help component.</p>
    </div>
)

export { HelpPage as default }