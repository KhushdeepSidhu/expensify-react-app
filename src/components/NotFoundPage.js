import React from 'react'

const NotFoundPage = () => (
    <div>
        <p>404! Page not found.</p>
        <Link to = "/">Go Home</Link>

    </div>
)

export { NotFoundPage as default }