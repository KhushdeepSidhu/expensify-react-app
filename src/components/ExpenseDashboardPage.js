import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashoardPage = () => (
    <div>
        <p>This is from the dashboard page component.</p>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export { ExpenseDashoardPage as default }