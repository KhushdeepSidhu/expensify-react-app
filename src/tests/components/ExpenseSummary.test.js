import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should render Expense summary correctly with 1 expense', () => {
    const wrapper = shallow ( 
        <ExpenseSummary
            expensesTotal = { 45600 }  
            expensesCount = { 1 }
        /> 
    ) 
    expect ( wrapper ).toMatchSnapshot ()
    
})

test('should render Expense summary correctly with multiple expenses', () => {
    const wrapper = shallow (
        <ExpenseSummary
            expensesTotal = { 345600 }
            expensesCount = { 3 }
        />
    )
    expect ( wrapper ).toMatchSnapshot ()
})
