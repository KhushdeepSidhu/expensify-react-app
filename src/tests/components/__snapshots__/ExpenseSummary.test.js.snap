// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense summary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling  
      <span>
        $456.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render Expense summary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling  
      <span>
        $3,456.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
